### PR TITLE
Added the ability to set OS X settings using defaults

### PR DESCRIPTION
**Pull request by opdude**

### CODE REVIEW

**Comment by jtand** [comment]
@opdude Thanks for this! Mind taking a look at [these lint failures](https://jenkins.saltstack.com/job/salt-pr-lint-n/12942/violations/)?

File: salt/states/mac_defaults.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    def safe_cast(val, to_type, default=None):
+        try:
+            return to_type(val)
+        except ValueError:
+            return default
+
+    current_value = __salt__['macdefaults.read'](domain, name, user)
+
+    if (type == 'bool' or type == 'boolean') and (
+                ((value is True or value is 'TRUE' or value is 'YES') and current_value == '1') or (
```
Comment (cachedout) [inline salt/states/mac_defaults.py]: There are a few issues here. First, we're overriding the use of the reserved keyword of `type` in Python, which should be avoided. Secondly, we should likely at least break this up into multiple conditionals for readability or consider the use of the `if x in [y,x]` syntax to make this easier to parse.

**Comment by cachedout** [comment]
@opdude There are some style concerns here. Could you please take a look at my comment in-line please? Thanks.

**Comment by cachedout** [comment]
Looks great now! Thanks, @opdude

**Pull request merged**